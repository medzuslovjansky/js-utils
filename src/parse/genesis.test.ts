import parseGenesis from './genesis';

describe('parseGenesis', () => {
  it.each([
    ['Arabic', 'A'],
    ['Deutsch', 'D'],
    ['English', 'E'],
    ['French', 'F'],
    ['German', 'G'],
    ['International', 'I'],
    ['Artificial', 'M'],
    ['Netherlands', 'N'],
    ['Oriental', 'O'],
    ['Slavic', 'S'],
    ['Turkic', 'T'],
  ])('should return %j for %j', (expected, input) => {
    expect(parseGenesis(input.toLowerCase())).toBe(expected);
    expect(parseGenesis(input.toUpperCase())).toBe(expected);
  });

  it.each([[null], ['Ar'], ['unknown']])(
    'should throw given an unknown value: %j',
    (input: any) => {
      expect(() => parseGenesis(input)).toThrowErrorMatchingSnapshot();
    },
  );
});
