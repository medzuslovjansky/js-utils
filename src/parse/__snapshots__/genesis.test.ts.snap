// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseGenesis should throw given an unknown value: "Ar" 1`] = `"Failed to parse word genesis: Ar"`;

exports[`parseGenesis should throw given an unknown value: "unknown" 1`] = `"Failed to parse word genesis: unknown"`;

exports[`parseGenesis should throw given an unknown value: null 1`] = `"Failed to parse word genesis: null"`;
